test/test_return.py

```diff
@@ -178,6 +178,26 @@ def test_return_location_edit(self):
             == "/#%23foo%3Fbar"
         ), 'location with a variable with encoding'
 
+        assert 'success' in self.conf(
+            '"https://${host}${uri}path$$sub"', 'routes/0/action/location'
+        ), 'location with $ sub-delimiter'
+        assert self.get()['headers']['Location'] == 'https://localhost/path$sub'
+
+        assert 'success' in self.conf(
+            '"https://$host$uri/$$sub1$$sub2"', 'routes/0/action/location'
+        ), 'location with 2 $ sub-delimiters'
+        assert self.get()['headers']['Location'] == 'https://localhost//$sub1$sub2'
+
+        assert 'success' in self.conf(
+            '"https://$host/routes$uri$method$$sub"', 'routes/0/action/location'
+        ), 'location with variables followed by a $ sub-delimiter'
+        assert self.get()['headers']['Location'] == 'https://localhost/routes/GET$sub'
+
+        assert 'success' in self.conf(
+            '"https://$host$uri$$$host"', 'routes/0/action/location'
```

Comment:
We could also add some test to check that `${$}` fails to configure.  If that behavior ever changes, we better know.

Reply:
@andrey-zelenkov are there pytests that are specific to variables instead of having to use some feature to test this?

Reply:
Another test that might be interesting: `${$host}` (should fail) :-)

Reply:
Yes, ${$host} fails, that bit of code is unchanged and bails out as soon as it sees a character other than [a-z_] between the curly braces.

A quick grep on the tests doesn't show a test for that case.

I'll add tests for those two failure cases, though they are technically the same failure case (non [a-z_] character after the '{'), but I'll wait until there's consensus on where the tests should go.

Reply:
Since these tests are not Location specific then it's better to move all these tests in the `test_variables_dollar()` method inside of `test_variables.py` (with all similar tests for $host, $uri, etc). test_static_variables.py is more about the tests related to the static feature while sub-delimiters look more general and related to the variables overall.

Reply:
Yes, this is more generally about being able to specify a '$' in strings which are processed for variables. I will move the tests into _test_variables.py_. Thanks!